pkg/cli/config/host.go

```diff
@@ -1,72 +1,130 @@
-		})
+	// Parse the URL
+	u, err := url.Parse(apiServerURL)
 	if err != nil {
-		return nil, err
+		return "", fmt.Errorf("failed to parse URL")
 	}
-	if len(serviceList.Items) == 0 {
-		return nil, errors.New("services for tekton results not found, try manual configuration")
+
+	hostname := u.Hostname()
+	if hostname == "" {
+		return "", errors.New("failed to extract hostname")
 	}
 
-	var routes []*v1.Route
-	for _, service := range serviceList.Items {
-		routeList, err := routeV1Client.Routes(service.Namespace).List(ctx, metav1.ListOptions{})
-		if err != nil {
-			return nil, err
-		}
-		if len(routeList.Items) == 0 {
-			return nil, errors.New("routes for tekton results not found, try manual configuration")
-		}
+	// For OpenShift/K8s, API server is typically: api.{cluster-domain}
+	// Extract {cluster-domain} part
+	if strings.HasPrefix(hostname, "api.") {
+		return strings.TrimPrefix(hostname, "api."), nil
+	}
 
-		for _, route := range routeList.Items {
-			if route.Spec.To.Name == service.Name {
-				port := route.Spec.Port.TargetPort
-				for _, p := range service.Spec.Ports {
-					if p.Port == port.IntVal || p.Name == port.StrVal {
-						routes = append(routes, &route)
-					}
-				}
-			}
+	// If it doesn't start with "api.", try to extract domain differently
+	// Handle cases like: k8s-api-server.cluster.example.com -> cluster.example.com
+	parts := strings.Split(hostname, ".")
+	if len(parts) >= 2 {
+		// Take the last two parts as domain (example.com)
+		// or more if it looks like a full domain
+		if len(parts) >= 3 {
+			return strings.Join(parts[1:], "."), nil // Skip first part
 		}
+		return strings.Join(parts, "."), nil
 	}
-	return routes, nil
+
+	return "", fmt.Errorf("unable to extract cluster domain")
+}
+
+// isURLReachable checks if a URL is reachable with a simple HTTP request
+func isURLReachable(testURL string) bool {
+	client := &http.Client{
+		// This HTTP client is only used for route discovery (connectivity testing),
+		// not for actual API communication. It allows insecure connections solely
+		// for the purpose of detecting service reachability in environments where
+		// certificates may not be trusted or available.
+		Timeout: 5 * time.Second,
+		// CodeQL[GoLangTLSDisabled] Safe because it's used only for test connection
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true}, // #nosec G402
+		},
+	}
+
+	resp, err := client.Get(testURL)
+	if err != nil {
+		return false
+	}
+	defer func() {
+		_ = resp.Body.Close() // Ignore close error for connectivity test
+	}()
+
+	// Consider any HTTP response (even errors like 401 Unauthorized, 403 Forbidden) as "reachable"
+	// because these indicate the service is present and responding, but may require authentication.
+	// This allows us to detect connectivity to the service even if credentials are missing or invalid.
+	return resp.StatusCode < 500 // 2xx, 3xx, 4xx (including auth errors) are all considered reachable
```

Comment:
Is a 500, 502, or 504 considered unreachable here? If I understand correctly that means the host _exists_ and we can _reach it_ but the Tekton Results API may be down. I think still tells us the URL is "reachable" in the sense we need here, but maybe that isn't suitable for other reasons?